fix: make findprovs return all responses

jacobheun opened the pull request:
Find providers was not consistently returning all of the provider responses. This was causing flakey test failures for https://github.com/libp2p/js-libp2p-delegated-content-routing/pull/9.

It will now properly return all response records for each result. 

* [x] requires https://github.com/ipfs/interface-js-ipfs-core/pull/493

alanshaw commented:
Code change LGTM - can we get a test in interface-ipfs-core that tests for multiple provider results? 🙏🙏🙏

jacobheun commented:
Added the findProvs test over at https://github.com/ipfs/interface-js-ipfs-core/pull/493. I also removed the skip here, as it wasnt running findProvs at all.
